Off-Top-App/off-top-java
# OT- 72 | Create a Post Request to Insert User Information

# Task Completed
- Created a post request in Springboot that would allow us to post user details into MYSQL database.

# Requirements for Testing
- Connection to MySQL database. How to set up MySQL-- https://github.com/Off-Top-App/off-top-java/pull/5
- Access to Postman. Download link https://www.postman.com/downloads/

# How to Test
**Testing in POSTMAN**
- Run the spring boot application. Then open postman and create a post-request using `http://localhost:9000/setUser`. Make sure `POST` request button is selected.
- Set up the following data in the body section under raw. Make sure we have `JSON` response selected next to GraphQL.  You can click the button which has Text response as default. 
`{
"Id": 1, 
"firstName": "John",
"lastName": "Snow",
"city": "WinterFell",
"age": 28, 
"gender": "Male",
"professional": "King in the North",
"email": "northKing01@gmail.com",
"username": "John_knows_everything",
"password": "HoldTheDoor",
"createdAt": "04/17/2011",
"deletedAt": "05/19/2019"
}`
- After you fill in the data click the send button It should look like the screen below in Postman.

![SQLpostman](https://user-images.githubusercontent.com/53457221/76155227-e537e900-609d-11ea-908c-61061732f7d5.JPG)

- Similarly the message `POST REQUEST CREATED SUCCESSFULLY` should be displayed in the console of the IDE.

![postrequest created](https://user-images.githubusercontent.com/53457221/76155349-e1a56180-609f-11ea-898d-2ddc21efa101.JPG)

**Testing MySQL Database**
- Open MySQL command line client in your computer. It should ask for a password and if you set up MySQL correctly `https://github.com/Off-Top-App/off-top-java/pull/5` it should be `password` and a database name `offTop` should be created in your local machine.
- Execute the SQL command `USE offTop` to access the database and `SELECT * FROM user` to access the user table in the database. The output should look like below.

![SQLcommandLine](https://user-images.githubusercontent.com/53457221/76155409-790ab480-60a0-11ea-8145-394eca30d704.JPG)

- If you see the above window then you have successfully posted the user information into the database.
